### PR TITLE
RRT Algorithm

### PR DESCRIPTION
**Pull request by shikha-16**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes!

### Description
Rapidly-exploring Random Trees for Robot Path Planning
A robot is attempting to navigate its path from the start point to a specified goal region, while avoiding the set of all obstacles.

In RRT, points are randomly generated within a specified radius and connected to the nearest existing node. Each time a node is created, we check that it lies outside of the obstacles. Furthermore, chaining the node to its closest neighbor must also avoid obstacles. The algorithm ends when a node is generated within the goal region, or a limit is hit

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#5138

### CODE REVIEW

**Comment by shikha-16** [comment]
@ankitaggarwal23 please review, the only checks that are failing are some minor errors in style

File: Python/Pathfinding/2D_RRT.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+#Purpose: A robot is attempting to navigate its path from the start point to a specified goal region, while avoiding the set of all obstacles.
+
+#Method: using tree structure
+
+#Intuition:
+#In RRT, points are randomly generated within a specified radius and connected to the nearest existing node in a tree. 
+#Each time a node is created, we check that it lies outside of the obstacles. 
+#Furthermore, chaining the node to its closest neighbor must also avoid obstacles. 
+#The algorithm ends when a node is generated within the goal region, or a limit is hit.
```
Comment (ankitaggarwal23) [inline Python/Pathfinding/2D_RRT.py:11]: Use Multi-Line Comments

File: Python/Pathfinding/2D_RRT.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+
+#have matplotlib and shapely installed
```
Comment (ankitaggarwal23) [inline Python/Pathfinding/2D_RRT.py:14]: Remove one extra line and redundant comments

File: Python/Pathfinding/2D_RRT.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    plt.figure(figsize=(15,10))
+    #define goal region here
+    Qgoal = [(90,90),(92,90),(92,92),(90,92)]
+
+    #define obstacles here
+    obstacles = [[(10,20),(10,95),(20,95),(20,20)],[(50,0),(50,80),(45,80),(45,0)],[(80,50),(80,90),(82,90),(82,50)]]
+
+    #number of iterations algorithm should run for
+    lim = 5000 
+
```
Comment (ankitaggarwal23) [inline Python/Pathfinding/2D_RRT.py:169]: Use dynamic input from user with proper display message

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
- Add sample I/O at the end of the Code
- Add Time and Space Complexity at the end of the Code   
- Add new line at the EOF

**Comment by shikha-16** [comment]
@ankitaggarwal23 please check out #5462

**Review by ankitaggarwal23 — COMMENTED** [review]
Found two opened by you. Closing this PR

**Pull request closed without merging**